frontend/src/components/DurationDropdown/DurationDropdown.tsx

```diff
@@ -47,6 +52,19 @@ export class DurationDropdown extends React.Component<DurationDropdownProps> {
       />
     );
   }
+
+  private updateDurationInterval = (duration: number) => {
+    this.props.setDuration(duration); // notify redux of the change
+
+    if (isKioskMode() ) {
+      const range = {rangeDuration: duration};
+      if (this.props.setTimeRange !== undefined) {
+        this.props.setTimeRange(range);
+      }
```

Comment:
"Duration" and "Time Ranges" are two different things. It is not OK to update both here.